scripts/apps/authoring/authoring/directives/AuthoringDirective.ts

```diff
@@ -1024,10 +1024,18 @@ export function AuthoringDirective(
             const removeListener = addInternalEventListener(
                 'dangerouslyOverwriteAuthoringData',
                 (event) => {
-                    if (event.detail._id === $scope.item._id) {
-                        angular.extend($scope.item, event.detail);
-                        angular.extend($scope.origItem, event.detail);
-                        $scope.$apply();
+                    if (event.detail.item._id === $scope.item._id) {
+                        angular.extend($scope.item, event.detail.item);
+
+                        if (event.detail.setDirty == false) {
+                            angular.extend($scope.origItem, event.detail.item);
```

Comment:
Yes, if setDirty is `undefined` which it is in most cases, this code would execute. In my "custom" case with the ai widget it wouldn't, which would result in a diff being created, so when I do $scope.dirty it would activate the Save button.